engine/framework/eventloop/runner.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eventloop
+
+import (
+	"context"
+	gerrors "errors"
+	"fmt"
+	"time"
+
+	"github.com/pingcap/errors"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/engine/pkg/clock"
+	derrors "github.com/pingcap/tiflow/engine/pkg/errors"
+)
+
+const (
+	defaultEventLoopTickInterval = 50 * time.Millisecond
+)
+
+// Runner runs a task.
+type Runner[T task] struct {
```

Comment:
Why we need a generic here?
Seems  eventloop of the framwork only cares about the common abilities of task, which means the base object can alread meet our need.

Reply:
I can't figure out what scenarios we will require a generic but not an interface `task` here.

Reply:
Because the DefaultBaseMaster and DefaultBaseWorker is a part of the framwork, which means we can control the implement of it.

Reply:
It does not seem necessary, but it can be good for readability, so that the reader of the code knows what the Runner runs.

Reply:
Ok. I hold my opinion.
@amyangfei @gozssky  PTAL~

Reply:
Maybe we can remove generic to make it simple, if generic is needed in the feature, we can add it back.